go/vt/vttablet/tabletmanager/vreplication/vplayer_flaky_test.go

```diff
@@ -2878,13 +2856,12 @@ func expectJSON(t *testing.T, table string, values [][]string, id int, exec func
 		if qr.Rows[i][0].ToString() != row[0] {
 			t.Fatalf("Id mismatch: want %s, got %s", qr.Rows[i][0].ToString(), row[0])
 		}
-		got, err := ajson.Unmarshal([]byte(qr.Rows[i][1].ToString()))
-		require.NoError(t, err)
-		want, err := ajson.Unmarshal([]byte(row[1]))
-		require.NoError(t, err)
-		match, err := got.Eq(want)
-		require.NoError(t, err)
-		require.True(t, match)
+
+		opts := jsondiff.DefaultConsoleOptions()
+		compare, s := jsondiff.Compare(qr.Rows[i][1].Raw(), []byte(row[1]), &opts)
+		if compare != jsondiff.FullMatch {
+			t.Errorf("Diff:\n%s\n", s)
+		}
```

Comment:
This gives us a far more superior error format when it fails and makes it much easier to debug what is going on.